app/controllers/foods_controller.rb

```diff
@@ -0,0 +1,76 @@
+class FoodsController < ApplicationController
+  load_and_authorize_resource
+  before_action :set_food, only: %i[show destroy]
+  # before_action :authorize_delete, only: [:destroy]
+  def index
+    @foods = Food.where(user_id: current_user.id)
+  end
+
+  def create(*recipe_id)
+    @user = current_user
+    recipe_id = params[:format]
+    @food = Food.new(params.require(:new_food).permit(:name, :measurement_unit, :price, :quantity))
+    @food.user = @user
+    if @food.save
+      flash[:notice] = 'Food created successfully!'
+      if recipe_id.nil?
+        redirect_to root_path
+      else
+        recipe_id = recipe_id.to_i
+        @recipe = Recipe.find(recipe_id)
+        @recipe_foods = RecipeFood.new(quantity: @food.quantity, food: @food, recipe: @recipe)
+        if @recipe_foods.save
+          flash[:notice] = 'Ingredient created successfully!'
+        else
+          flash[:alert] = @recipe_foods.errors.full_messages.join(', ')
+        end
+        redirect_to recipe_path(id: recipe_id)
+      end
+    else
+      flash[:alert] = @food.errors.full_messages.join(', ')
+      food = Food.new
+      # format.html { redirect_to request.referrer, locals: { food: food } }
+      redirect_to new_food_path, locals: { food: }
```

Comment:
- [ ] I've got an error when I start the page. I'm not sure if you got an error or not, but when I add there 

redirect_to new_food_path, locals: { food: **@food** }, then it fixed.

![Screenshot from 2023-02-10 21-34-33](https://user-images.githubusercontent.com/26767383/218141917-a221a36a-b1a5-44b1-837f-5e98a694c0a2.png)

Please fix that.